core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/dao/JPAAuditDAO.java

```diff
@@ -19,50 +19,88 @@
 import org.apache.syncope.core.persistence.api.dao.AuditDAO;
-import org.apache.syncope.core.persistence.api.dao.search.AnyCond;
-import org.apache.syncope.core.persistence.api.dao.search.AttributeCond;
 import org.apache.syncope.core.persistence.api.dao.search.OrderByClause;
-import org.apache.syncope.core.persistence.api.dao.search.SearchCond;
 import org.apache.syncope.core.persistence.api.entity.AuditEntry;
+import org.apache.syncope.core.provisioning.api.AuditEntryImpl;
+import org.apache.syncope.core.provisioning.api.serialization.POJOHelper;
+import org.apache.syncope.core.spring.security.AuthContextUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Repository;
 import org.springframework.transaction.annotation.Transactional;
 
+import javax.sql.DataSource;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.Date;
 import java.util.List;
-import java.util.Set;
+import java.util.stream.Collectors;
 
 @Transactional(rollbackFor = Throwable.class)
 @Repository
-public class JPAAuditDAO implements AuditDAO<AuditEntry> {
+public class JPAAuditDAO extends AbstractDAO implements AuditDAO<AuditEntry> {
     private static final Logger LOG = LoggerFactory.getLogger(JPAAuditDAO.class);
 
-    @Override
-    public AuditEntry findByUsername(final String who) {
-        return null;
-    }
+    @Autowired
+    private DomainsHolder domainsHolder;
 
     @Override
-    public SearchCond getAllMatchingCond() {
-        AnyCond idCond = new AnyCond(AttributeCond.Type.ISNOTNULL);
-        idCond.setSchema("logger");
-        return SearchCond.getLeafCond(idCond);
+    public List<AuditEntry> findByEntityKey(final String key, final int page,
+                                            final int itemsPerPage,
+                                            final List<OrderByClause> orderByClauses) {
+        try {
+            String queryString = "SELECT * FROM SYNCOPEAUDIT WHERE MESSAGE LIKE '%" + key + "%' ";
+            if (!orderByClauses.isEmpty()) {
+                 queryString += " ORDER BY " + orderByClauses.
+                    stream().
+                    map(orderBy -> orderBy.getField() + " " + orderBy.getDirection().name()).
+                    collect(Collectors.joining(","));
+            }
+            JdbcTemplate template = getJdbcTemplate();
+            template.setMaxRows(itemsPerPage);
+            template.setFetchSize(itemsPerPage * (page <= 0 ? 0 : page - 1));
+            return template.query(queryString, new AuditEntryRowMapper(key));
+        } catch (SyncopeClientException e) {
+            throw e;
+        } catch (Exception e) {
+            LOG.error(e.getMessage(), e);
```

Comment:
Can you add some context to the log message?

Reply:
Updated error message to be clearer